beginner_source/torchtext_custom_dataset_tutorial.py

```diff
@@ -0,0 +1,383 @@
+# -*- coding: utf-8 -*-
+"""
+Pre-process custom text dataset using Torchtext
+===============================================
+
+**Author**: `Anupam Sharma <https://anp-scp.github.io/>`_
+
+This tutorial illustrates the usage of torchtext on a dataset that is not built-in. In the tutorial,
+we will pre-process a dataset that can be further utilized to train a sequence-to-sequence
+model for machine translation (something like, in this tutorial: `Sequence to Sequence Learning
+with Neural Networks <https://github.com/bentrevett/pytorch-seq2seq/blob/master/1%20-%20Sequence%\
+20to%20Sequence%20Learning%20with%20Neural%20Networks.ipynb>`_) but without using legacy version
+of torchtext.
+
+In this tutorial, we will learn how to:
+
+* Read a dataset
+* Tokenize sentence
+* Apply transforms to sentence
+* Perform bucket batching
+
+Let us assume that we need to prepare a dataset to train a model that can perform English to
+German translation. We will use a tab-delimited German - English sentence pairs provided by
+the `Tatoeba Project <https://tatoeba.org/en>`_ which can be downloaded from this link: `Click
+Here <https://www.manythings.org/anki/deu-eng.zip>`__.
+
+Sentence pairs for other languages can be found in this link:
```

Comment:
nit: same comment as above. Let's hyperlink the "this link" text and get rid of the line below.

Reply:
I have made changes as suggested.